modules/node_groups/launchtemplate.tf

```diff
@@ -0,0 +1,111 @@
+data "cloudinit_config" "workers_userdata" {
+  for_each      = { for k, v in local.node_groups_expanded : k => v if v["create_launch_template"] }
+  gzip          = false
+  base64_encode = true
+  boundary      = "//"
+
+  part {
+    content_type = "text/x-shellscript"
+    content = templatefile("${path.module}/templates/userdata.sh.tpl",
+      {
+        pre_userdata       = each.value["pre_userdata"]
+        kubelet_extra_args = each.value["kubelet_extra_args"]
+      }
+    )
+
+  }
+}
+
+# This is based on the LT that EKS would create if no custom one is specified (aws ec2 describe-launch-template-versions --launch-template-id xxx)
+# there are several more options one could set but you probably dont need to modify them
+# you can take the default and add your custom AMI and/or custom tags
+#
+# Trivia: AWS transparently creates a copy of your LaunchTemplate and actually uses that copy then for the node group. If you DONT use a custom AMI,
+# then the default user-data for bootstrapping a cluster is merged in the copy.
+resource "aws_launch_template" "workers" {
+  for_each               = { for k, v in local.node_groups_expanded : k => v if v["create_launch_template"] }
+  name_prefix            = lookup(each.value, "name", join("-", [var.cluster_name, each.key, random_pet.node_groups[each.key].id]))
+  description            = lookup(each.value, "name", join("-", [var.cluster_name, each.key, random_pet.node_groups[each.key].id]))
+  update_default_version = true
+
+  block_device_mappings {
+    device_name = "/dev/xvda"
+
+    ebs {
+      volume_size           = lookup(each.value, "disk_size", null)
+      volume_type           = lookup(each.value, "disk_type", null)
+      delete_on_termination = true
+    }
+  }
+
+  instance_type = each.value.instance_type
```

Comment:
When I attempt to build with a simple `node_groups` such as: 
```
node_groups = {
    general_pods = {
      instance_types = ["m4.large", "t3a.large"]
      desired_capacity = 1
      min_capacity = 1
      max_capacity = 5
      capacity_type = "SPOT"
      create_launch_template = true
      kubelet_extra_args = "--node-labels=lifecycle=Ec2Spot --register-with-taints=Ec2Spot=true:NoSchedule",

      tags = {
        "k8s.io/cluster-autoscaler/node-template/label/lifecycle" = "Ec2Spot"
        "k8s.io/cluster-autoscaler/node-template/taint/Ec2Spot" = "true:NoSchedule"
        "k8s.io/cluster-autoscaler/node-template/resources/cpu" = "2"
        "k8s.io/cluster-autoscaler/node-template/resources/memory": "8Gi"
      }
    }
}
```
It fails by not being able to get the instance_types:
```
Error: Unsupported attribute

  on ../../../../../terraform-aws-eks/modules/node_groups/launchtemplate.tf line 41, in resource "aws_launch_template" "workers":
  41:   instance_type = each.value.instance_type
    |----------------
    | each.value is object with 20 attributes

This object does not have an attribute named "instance_type".
```
Is there something I'm misunderstanding about how this should be initialized?

Reply:
The issue is that it appears as though we cannot use the `instance_types` attribute as defined. Only when I specifically say `instance_type` and use only one will it work.

Reply:
Not sure I have updated to 14.0 so that might be the issue

Reply:
Should be fixed with latest commit